src/com/gmail/alinakotova102/drafts/StepArrays.java

```diff
@@ -1,7 +1,251 @@
 package com.gmail.alinakotova102.drafts;
 
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.concurrent.ThreadLocalRandom;
+
 public class StepArrays {
-    public static void main(String[] args){
+    public static Integer NOT_DEFINE = null;
+
+    public static void main(String[] args) {
+        System.out.println("Creation of a stepped array with random numbers");
+        System.out.print("Enter N - the number of lines: ");
+        int countRows = enterNumberBetween();
+        System.out.print("Enter M - the maximum number of elements in a row: ");
+        int maxCount = enterNumberBetween();
+
+        System.out.print("\nEnter number for create array with random numbers\nMin number: ");
+        int minRandom = enterNumber();
+        System.out.print("Max number: ");
+        int maxRandom = enterNumber();
+        while (Math.abs(maxRandom - minRandom) == 1 || maxRandom == minRandom) {
+            System.out.println("Error! There is no range between numbers\nMax number: ");
+            maxRandom = enterNumber();
+        }
+
+        int[][] array = createArray(countRows, maxCount);
+        array = fillArray(array, minRandom, maxRandom);
+
+        System.out.println("\nInitial array");
+        display(array);
+
+        System.out.println("Modified array");
+        int[][] arraySorted = sortEvenOdd(array);
+        display(arraySorted);
+
+        int sumElements = sumElements(array);
+        System.out.println("Sum of elements: " + sumElements);
+
+        Integer[] minElement = minElementInRow(array);
+        System.out.print("\nMinimal element in row: ");
+        display(minElement);
+
+        int absoluteMinElement = absoluteMin(array);
+        System.out.println("\nAbsolute minimal element: " + absoluteMinElement);
```

Comment:
```java
 public static Integer absoluteMin(int[][] matrix) {
        if (min(minElementInRow(matrix)) == 0) {
            return NOT_DEFINE;
        }
        return min(minElementInRow(matrix));
    }
``` 
![image](https://github.com/AlinaKlishyna/KlishynaAlina_JavaBase_03_02_2024/assets/48547022/1683b0ef-e1df-4ff0-b6c3-6903e0958984)

Reply:
Я запустив твій код з гілки main і отримав NPE:
```
Creation of a stepped array with random numbers
Enter N - the number of lines: 4
Enter M - the maximum number of elements in a row: 1

Enter number for create array with random numbers
Min number: 0
Max number: 2

Initial array
[]
[]
[]
[]

Modified array
[]
[]
[]
[]

Sum of elements: 0

Minimal element in row: [null, null, null, null]
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because "minValue" is null
	at com.gmail.alinakotova102.drafts.StepArrays.min(StepArrays.java:82)
	at com.gmail.alinakotova102.drafts.StepArrays.absoluteMin(StepArrays.java:65)
	at com.gmail.alinakotova102.drafts.StepArrays.main(StepArrays.java:43)
```